go/protocol/chat1/extras.go

```diff
@@ -340,6 +341,20 @@ func (m MessageUnboxedValid) AsDeleteHistory() (res MessageDeleteHistory, err er
 	return m.MessageBody.Deletehistory(), nil
 }
 
+func (m MessageUnboxedValid) GetEphemeralMetadata() *MsgEphemeralMetadata {
+	return m.ClientHeader.EphemeralMetadata
+}
+
+func (m MessageUnboxedValid) IsEphemeralExpired() bool {
+	metadata := m.GetEphemeralMetadata()
+	expired := time.Now().Sub(m.ServerHeader.Ctime.Time()) >= time.Second*time.Duration(metadata.Lifetime)
```

Comment:
`time.Now()` seems like it's going to bring in clock skew problems here. Is there any way to pass some merkle tree info down?

Reply:
Is that really relevant in a world where devices can be offline anyway?

Reply:
I guess my big worry is that clock skew causes one device to delete key X, while another device thinks it's still alive and keeps encrypting for it. Deleting individual messages seems like less of an issue, though, if it makes sense to use a different mechanism.

Reply:
let's discuss in person (this isn't used in this PR but https://github.com/keybase/client/pull/11347 anyway) but on the server side when we assign the etime we just use the server's view of now so i don't think we should be using merkle times here (but definitely for key deletion as we do). maybe we can send the server's `ktime` down and use the `max(now(), ktime)` so if a client is behind the server it cleans up things that the server has